iterator/src/test/java/com/iluwatar/iterator/list/TreasureChestTest.java

```diff
@@ -60,13 +55,14 @@
   }
 
   /**
-   * Test if the expected item can be retrieved from the chest using the {@link ItemIterator}
+   * Test if the expected item can be retrieved from the chest using the {@link
+   * TreasureChestItemIterator}
    */
   @ParameterizedTest
   @MethodSource("dataProvider")
   public void testIterator(Item expectedItem) {
     final TreasureChest chest = new TreasureChest();
-    final ItemIterator iterator = chest.iterator(expectedItem.getType());
+    final TreasureChestItemIterator iterator = chest.iterator(expectedItem.getType());
```

Comment:
Same change here. You being the client of `TreasureChest` don't care if the concrete implementation of `Iterator` is `TreasureChestItemIterator` or something else. You just need an `Iterator`. This allows for easier refactoring as you can freely change name of the class and client code will not be affected. I hope that makes sense.

Reply:
Ah, crap. Good catch!